src/components/candevice/candevice.cpp

```diff
@@ -34,6 +44,19 @@ bool CanDevice::init(const QString& backend, const QString& interface)
     return d->_initialized;
 }
 
+bool CanDevice::init()
+{
+    Q_D(CanDevice);
+    if (d->_props.count(d->_backendProperty) != 1 ||
```

Comment:
Duplicated condition here.

Reply:
Good that you noticed this, it is actually a bug not a duplicate condition, _interfaceProperty should be the second part of the condition and also second parameter few lines before this